File: samples_c/s61_utf8_byte_vs_char_count.c
#include <stdio.h>

/* Girdi: tek satir metin (UTF-8 olabilir, Turkce karakterler icerebilir).
   Cikti: "<bayt_sayisi> <karakter_sayisi>".
   NOT: Kok Neden B (string modeli) icin 3. bagimsiz ornek (bkz. s06, s13).
   Bayt sayisi ile UTF-8 kod noktasi sayisi cok baytli girdide farklidir;
   "dogal" bir Rust cevirisi .len() (bayt) ile .chars().count() (kod noktasi)
   arasindaki farki gozden kacirabilir. */
int main(void) {
    char buf[4096];
    if (fgets(buf, sizeof(buf), stdin) == NULL) return 0;
    int bytes = 0, chars = 0;
    for (int i = 0; buf[i] != '\0'; i++) {
        if (buf[i] == '\n' || buf[i] == '\r') break;
        unsigned char c = (unsigned char)buf[i];
        bytes++;
        if ((c & 0xC0) != 0x80) {
            chars++;
        }
    }
    printf("%d %d\n", bytes, chars);
    return 0;
}

File: translations_rust__gemini/s61_utf8_byte_vs_char_count.rs
use std::io::{self, Read};

fn main() {
    let mut buf = [0u8; 4096];
    let mut len = 0;
    let stdin = io::stdin();
    let mut handle = stdin.lock();

    while len < 4095 {
        let mut byte = [0u8; 1];
        match handle.read(&mut byte) {
            Ok(1) => {
                buf[len] = byte[0];
                len += 1;
                if byte[0] == b'\n' {
                    break;
                }
            }
            _ => break,
        }
    }

    if len == 0 {
        return;
    }

    let mut bytes = 0;
    let mut chars = 0;
    for &c in &buf[..len] {
        if c == b'\0' || c == b'\n' || c == b'\r' {
            break;
        }
        bytes += 1;
        if (c & 0xC0) != 0x80 {
            chars += 1;
        }
    }

    println!("{} {}", bytes, chars);
}
